framework/src/media/audio/audio_manager.c

```diff
@@ -883,18 +985,34 @@ int start_audio_stream_in(void *data, unsigned int frames)
 				frames_to_read = (get_input_frame_count() / card->resample.ratio);
 			}
 
-			card->resample.buffer_size = get_input_frames_to_byte(frames);
+			card->resample.buffer_size = get_user_input_frames_to_byte(frames) / rechanneling_ratio;
 
 			ret = pcm_readi(card->pcm, card->resample.buffer, frames_to_read);
-			medvdbg("Read frames (%d/%u) for resample to %u\n", ret, frames_to_read, frames);
+			medvdbg("Read frames (%d/%u) in %u for resampling to %u\n", ret, frames_to_read, card->resample.buffer_size, frames);
 		} else {
-			ret = pcm_readi(card->pcm, data, frames);
+			if (rechanneling_ratio < 1) {	// Stereo -> Mono
```

Comment:
we can remove card->resample.necessary too

Reply:
Removed.